site/content/contribute/desktop/_index.md

```diff
@@ -6,16 +6,46 @@ date: 2018-01-02T10:36:34-05:00
 weight: 7
 ---
 
-The Mattermost desktop app is an [Electron](https://electronjs.org/) wrapper around the [web app](/contribute/webapp) project. It lives in the [mattermost/desktop](https://github.com/mattermost/desktop) repository. The desktop app runs on Windows, Linux, and macOS.
+The Mattermost Desktop App is an [Electron](https://electronjs.org/) wrapper around the [web app](/contribute/webapp) project. It lives in the [mattermost/desktop](https://github.com/mattermost/desktop) repository. The desktop app runs on Windows, Linux, and macOS.
 
-## Repository
+#### Electron
+The Desktop App, like all Electron apps is broken into two pieces, the **main** process and the **renderer** process.
 
-You can find the repository [here](https://github.com/mattermost/desktop).
+The **main** process is a NodeJS process that has access to operating system functions, and governs the creation and management of several renderer processes.  
+The **renderer** processes are Chromium instances that perform different functions. In our app, each Mattermost server is its own renderer process.
 
-## Channel
+![Process diagram](/contribute/desktop/process-diagram.png)
 
-You can find the channel [here](https://community.mattermost.com/core/channels/desktop-app).
+In order to facilitate communication between the two processes, there is a communication layer in which information can be sent between. We expose *ONLY* the communication API to the renderer process so that we don't allow any malicious server to wreak havoc on a user's computer.
+
+You can read more about the Process Model [here](https://www.electronjs.org/docs/latest/tutorial/process-model).
+
+#### Directory Structure
+The directory structure is broken down into a few pieces to best organize the code:
+
+```
+Mattermost Desktop
+├── docs/ - Documentation for working on the Desktop App
+├── e2e/ - E2E tests
+│   ├── modules/ - Setup code for the E2E tests
+│   └── specs/ - E2E tests themselves
+├── resources/ - Assets, such as images or sound files that the Desktop App uses
```

Comment:
```suggestion
├── resources/ - Assets such as images or sound files that the Desktop App uses
```

Reply:
Curious: what's the main difference between the files in **resources** and the ones in **assets**?